source/structured-streaming.txt

```diff
@@ -0,0 +1,276 @@
+.. _spark-structured-streaming:
+
+=================================
+Structured Streaming with MongoDB
+=================================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Spark Structured Streaming is a data stream processing engine you can 
+use through the Dataset or DataFrame API. The MongoDB Spark Connector 
+lets you stream to and from MongoDB using Spark Structured Streaming.
+
+.. include:: includes/streaming-distinction.rst
+
+To learn more about Structured Streaming, see the 
+`Spark Programming Guide
+<https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html>`__.
+
+.. _write-structured-stream:
+
+Configuring a Write Stream to MongoDB
+-------------------------------------
+
+Specify ``.writeStream`` configuration settings on your streaming 
+Dataset or DataFrame.
+
+.. tabs-drivers::
+
+   tabs:
+     - id: java-sync
+       content: |
+
+     - id: python
+       content: |
+
+         .. code-block:: python
+            :copyable: true
+         
+            <streaming Dataset/ DataFrame> \
+              .writeStream \
+              .queryName("exampleQuery") \
+              .format("mongodb") \
+              .option("checkpointLocation", "/tmp/checkpoint/") \
+              .option("forceDeleteTempCheckpointLocation", "true") \
+              .option("spark.mongodb.connection.uri", <mongodb-connection-string>) \
+              .outputMode("append")
+
+         For a complete list of methods, see the 
+         `pyspark Structured Streaming reference <https://spark.apache.org/docs/latest/api/python/reference/pyspark.ss.html>`__.
+
+     - id: scala
+       content: |
+
+.. _read-structured-stream:
+.. _continuous-processing:
+
+Configuring a Read Stream from MongoDB
+--------------------------------------
+
+Reading a stream from a MongoDB database requires 
+*continuous processing*, 
+an experimental feature introduced in Spark version 2.3. To learn 
+more about continuous processing, see the `Spark documentation <https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#continuous-processing>`__.
+
+To use continuous processing with the MongoDB Spark Connector, add the 
+``.trigger()`` method to the ``.writeStream`` of the streaming 
+Dataset or DataFrame that you create from your MongoDB ``.readStream``.
+
+.. note:: 
+
+   Reading a stream from MongoDB with continuous processing depends 
+   upon watching :manual:`change streams </changeStreams>` for 
+   real-time updates.
+
+   To read a stream from MongoDB, you must stream data to MongoDB.
```

Comment:
Issue: I can't tell what the purpose of this note is. If it is saying that you must update data in MongoDB in order to populate your change stream, could that be explicitly stated?

I'm not sure what is meant by "stream data to MongoDB", as I believe a database's change stream can be populated without a streaming engine like Kafka or Spark Structured Streaming (I believe you can update data through standard CRUD operations to populate the change stream).

```suggestion
.. note:: 

   Your connector populates it's read stream from your MongoDB 
   deployment's change stream. To populate your change stream, perform
   update operations on your MongoDB database.
   
   To learn more about change streams, see :manual:`Change Streams </changeStreams>`
   in the MongoDB manual.
```

Reply:
>  If it is saying that you must update data in MongoDB in order to populate your change stream, could that be explicitly stated?

Yes that's it, I'll clarify